app/src/androidTest/java/com/github/epfl/meili/review/ReviewsActivityTest.kt

```diff
@@ -0,0 +1,314 @@
+    }
+
+    @Test
+    fun signedInEditReviewTest() {
+        mockAuthenticationService.signInIntent()
+        service.onEvent(mockSnapshotAfterAddition, null) // mock user has existing review
+
+        onView(withId(R.id.fab_add_edit_review)).perform(click())
+
+        // old review is displayed
+        onView(withId(R.id.edit_title)).check(matches(editTextContainsText(TEST_ADDED_TITLE)))
+        onView(withId(R.id.edit_summary)).check(matches(editTextContainsText(TEST_SUMMARY)))
+        onView(withId(R.id.rating_bar)).check(matches(ratingBarHasRating(ADDED_REVIEW_RATING)))
+
+        // edit review
+        onView(withId(R.id.rating_bar)).perform(setRating(EDITED_REVIEW_RATING))
+        onView(withId(R.id.edit_title)).perform(clearText(), typeText(TEST_EDITED_TITLE), closeSoftKeyboard())
+        onView(withId(R.id.submit_review)).perform(click())
+
+        // send reviews map with edited review to review service
+        service.onEvent(mockSnapshotAfterEdition, null)
+
+        onView(withId(R.id.list_reviews)).check(matches(isDisplayed()))
+        onView(withId(R.id.edit_review)).check(matches(not(isDisplayed())))
+
+        onView(withId(R.id.recycler_view))
+            .check(matches(isDisplayed()))
+            .perform(RecyclerViewActions.scrollTo<RecyclerView.ViewHolder>(hasDescendant(withText(TEST_UID))))
+
+        onView(withText(TEST_UID)).check(matches(isDisplayed()))
+        onView(textViewContainsText(TEST_EDITED_TITLE)).check(matches(isDisplayed()))
+
+        onView(withText(TEST_ADDED_TITLE)).check(doesNotExist())
+
+        onView(withId(R.id.average_rating)).check(matches(textViewContainsText(AVERAGE_FORMAT.format(testAverageRatingAfterEdition))))
+    }
+
+    private fun ratingBarHasRating(rating: Float): Matcher<View?> {
+        return object : TypeSafeMatcher<View?>() {
+            override fun describeTo(description: Description) {
+                description.appendText("A RatingBar with the value: $rating")
+            }
+
+            override fun matchesSafely(item: View?): Boolean {
+                when (item) {
+                    is RatingBar -> return item.rating == rating
+                }
+                return false
+            }
+        }
+    }
+
+    private fun textViewContainsText(content: String): Matcher<View?> {
+        return object : TypeSafeMatcher<View?>() {
+            override fun describeTo(description: Description) {
+                description.appendText("A TextView with the text: $content")
+            }
+
+            override fun matchesSafely(item: View?): Boolean {
+                when (item) {
+                    is TextView -> return item.text == content
+                }
+                return false
+            }
+        }
+    }
+
+    private fun editTextContainsText(content: String): Matcher<View?> {
+        return object : TypeSafeMatcher<View?>() {
```

Comment:
Could you please tell me how I should avoid this repetition? Since it's a constructor call, I don't see another way to initialise an object without calling its constructor